src/wizard.tsx

```diff
@@ -16,6 +23,12 @@ const Wizard: React.FC<React.PropsWithChildren<WizardProps>> = React.memo(
     hasNextStep.current = activeStep < stepCount - 1;
     hasPreviousStep.current = activeStep > 0;
 
+    React.useEffect(() => {
```

Comment:
Not what useEffect is made for. Rather create a handleStepChangeFunction which includes the setActiveStep and replace all setActiveStep calls with handleStepChangeFunction. Than add onStepChange/onIndexChange tot that function.

Reply:
Indeed. We already keep track of the active index in the wizard. So let's extend the `goToPreviousStep` and `goToNextStep` methods and invoke the `onStepChange` callback